src/rust/ide/lib/utils/src/test.rs

```diff
@@ -97,3 +97,24 @@ for Collection {
         (v1,v2,v3,v4,v5)
     }
 }
+
+#[allow(clippy::type_complexity)]
+impl<Collection: IntoIterator> ExpectTuple
+    <(Collection::Item,Collection::Item,Collection::Item,Collection::Item,Collection::Item
+    ,Collection::Item)>
+for Collection {
+    fn expect_tuple
+    (self)
+     -> (Collection::Item,Collection::Item,Collection::Item,Collection::Item,Collection::Item
+            ,Collection::Item) {
+        let mut iter = self.into_iter();
+        let     v1   = iter.next().unwrap();
+        let     v2   = iter.next().unwrap();
+        let     v3   = iter.next().unwrap();
+        let     v4   = iter.next().unwrap();
+        let     v5   = iter.next().unwrap();
+        let     v6   = iter.next().unwrap();
+        assert!(iter.next().is_none());
+        (v1,v2,v3,v4,v5,v6)
```

Comment:
btw this sohuld be generated with macro - it can be done very very simly (I'd love to help!)